File: secretmanager/v1/python/google-cloud-secret-manager/secret-manager-client-secret-get.py
# Copyright 2026 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# [START secretmanager_v1_secretmanagerservice_secret_get]
import google.api_core.exceptions
from google.cloud import secretmanager_v1


def get_secret_metadata(project_id: str, secret_id: str) -> None:
    """Retrieves the metadata for a specific secret. This demonstrates how to fetch information about a secret resource without accessing its payload.

    Args:
        project_id: Google Cloud Project ID (such as 'example-project-id')
        secret_id: ID of the secret to retrieve (such as 'my-secret-id')
    """
    client = secretmanager_v1.SecretManagerServiceClient()

    name = f"projects/{project_id}/secrets/{secret_id}"

    try:
        request = secretmanager_v1.GetSecretRequest(
            name=name,
        )

        secret = client.get_secret(request=request)

        print(f"Found secret: {secret.name}")
        print(f"  Create Time: {secret.create_time}")
        print(f"  Labels: {secret.labels}")
        print(f"  Replication: {secret.replication}")

    except google.api_core.exceptions.NotFound:
        print(
            f"Error: The secret '{secret_id}' was not found in project "
            f"'{project_id}'. Verify the secret exists and the "
            "name is correct."
        )
    except Exception as e:
        print(f"An unexpected error occurred: {e}")


# [END secretmanager_v1_secretmanagerservice_secret_get]

File: secretmanager/v1/nodejs/google-cloud-secret-manager/secret-manager-client-secret-get.js
/**
 * Copyright 2026 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// [START secretmanager_v1_secretmanagerservice_secret_get]
/**
 * @param projectId Google Cloud Project ID (such as 'example-project-id')
 * @param secretId ID of the secret to retrieve (such as 'my-secret-id')
 */
'use strict';

const {SecretManagerServiceClient} = require('@google-cloud/secret-manager');
const {status} = require('@grpc/grpc-js');

const client = new SecretManagerServiceClient();

async function getSecretMetadata(projectId, secretId) {
  const name = `projects/${projectId}/secrets/${secretId}`;

  try {
    const [secret] = await client.getSecret({
      name,
    });

    const replication = secret.replication;
    let replicationInfo;
    if (replication.automatic) {
      replicationInfo = 'automatic';
    } else if (replication.userManaged) {
      replicationInfo = 'user-managed';
    } else {
      replicationInfo = 'unknown';
    }

    const dateObj = new Date(Number(secret.createTime.seconds) * 1000);

    const createTime = new Intl.DateTimeFormat('en-US', {
      dateStyle: 'short',
      timeStyle: 'long',
      timeZone: 'GMT',
    }).format(dateObj);

    console.log(`Found secret: ${secret.name}`);
    console.log(`  Create Time: ${createTime}`);
    console.log(`  Labels: ${JSON.stringify(secret.labels)}`);
    console.log(`  Replication: ${replicationInfo}`);
  } catch (err) {
    if (err.code === status.NOT_FOUND) {
      console.error(
        `Error: The secret '${name}' was not found. Verify the secret exists and the name is correct.`,
      );
    } else {
      console.error(`An unexpected error occurred: ${err.message}`);
    }
  }
}

module.exports = {getSecretMetadata};
// [END secretmanager_v1_secretmanagerservice_secret_get]
